### PR TITLE
release: v1.40.0

### PR DESCRIPTION
**Pull request by abernix**
> **Note**
> **This particular PR must be true-merged to `main`.**

* This PR is only ready to review when it is marked as "Ready for Review".  It represents the merge to the `main` branch of an upcoming release (version number in the title).
* It will act as a staging branch until we are ready to finalize the release.
* We may cut any number of alpha and release candidate (RC) versions off this branch prior to formalizing it.
* This PR is **primarily a merge commit**, so reviewing every individual commit shown below is **not necessary** since those have been reviewed in their own PR.  However, things important to review on this PR **once it's marked "Ready for Review"**:
    - Does this PR target the right branch? (usually, `main`)
    - Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, "Did the 'release prep' PR actually land on this branch?"
    - If those things look good, this PR is good to merge!

### CODE REVIEW

File: .changesets/feat_lrlna_remove_header_in_rhai.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+### Add a .remove method for headers in Rhai
```
Comment (abernix) [inline .changesets/feat_lrlna_remove_header_in_rhai.md:1]:
```suggestion
### Add a `.remove` method for headers in Rhai
```

File: .changesets/fix_fix_active_session_count_on_dropped_futures.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+### Fix active_session_count when future is dropped ([Issue #4601](https://github.com/apollographql/router/issues/4601))
```
Comment (abernix) [inline .changesets/fix_fix_active_session_count_on_dropped_futures.md:1]:
```suggestion
### Fix `active_session_count` when future is dropped ([Issue #4601](https://github.com/apollographql/router/issues/4601))
```

File: .changesets/istio_virtualservice_gateway_list.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+### Helm update to allow a list of gateways to VirtualService ([Issue #4464](https://github.com/apollographql/router/issues/4464))
+
+Allow to configure 0..N gateways in Istio VirtualService 
+
+Fixes [#4464](https://github.com/apollographql/router/issues/4464)
+
```
Comment (abernix) [inline .changesets/istio_virtualservice_gateway_list.md:6]:
```suggestion
Fixes [#4464](https://github.com/apollographql/router/issues/4464)

```

File: .changesets/feat_lrlna_remove_header_in_rhai.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+This allows users to remove headers in a Rhai script with a `.remove`. For example:
```
Comment (shorgi) [inline .changesets/feat_lrlna_remove_header_in_rhai.md:3]:
```suggestion
The router supports a new `.remove` method that enables users to remove headers in a Rhai script. 

For example:
```

File: .changesets/istio_virtualservice_gateway_list.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+### Helm update to allow a list of gateways to VirtualService ([Issue #4464](https://github.com/apollographql/router/issues/4464))
+
+Allow to configure 0..N gateways in Istio VirtualService
```
Comment (shorgi) [inline .changesets/istio_virtualservice_gateway_list.md:3]: Configuration of the router's Helm chart has been updated to allow multiple gateways. This enables configuration of multiple gateways in Istio VirtualService.

The previous configuration for a single `virtualservice.gatewayName` has been deprecated in favor of a configuration for an array of `virtualservice.gatewayNames`.

Comment (shorgi) [inline .changesets/istio_virtualservice_gateway_list.md:3]:
```suggestion
Configuration of the router's Helm chart has been updated to allow multiple gateways. This enables configuration of multiple gateways in Istio VirtualService.

The previous configuration for a single `virtualservice.gatewayName` has been deprecated in favor of a configuration for an array of `virtualservice.gatewayNames`.
```

**Review by o0Ignition0o — APPROVED** [review]

**Pull request merged**